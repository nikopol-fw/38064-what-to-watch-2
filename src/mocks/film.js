export const film = {
  title: `The Grand Budapest Hotel`,
  genre: `Drama`,
  releaseYear: `2014`,
  posterImg: `bg-the-grand-budapest-hotel.jpg`,
  coverImg: `the-grand-budapest-hotel-poster.jpg`,
};
