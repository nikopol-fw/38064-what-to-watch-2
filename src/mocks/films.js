export const films = [{
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  img: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
}, {
  title: `Bohemian Rhapsody`,
  img: `bohemian-rhapsody.jpg`,
}, {
  title: `Macbeth`,
  img: `macbeth.jpg`,
}, {
  title: `Aviator`,
  img: `aviator.jpg`,
}, {
  title: `We need to talk about Kevin`,
  img: `we-need-to-talk-about-kevin.jpg`,
}];
