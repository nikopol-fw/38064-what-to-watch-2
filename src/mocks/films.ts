import {Film} from "../models/Film";


export const films: Film[] = [{
  backgroundColor: `#A6B7AC`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/gangs_of_new_york.jpg`,
  description: `In 1862, Amsterdam Vallon returns to the Five Points area of New York City seeking revenge against Bill the Butcher, his father's killer.`,
  director: `Martin Scorsese`,
  genre: `Crime`,
  id: 1,
  isFavorite: false,
  name: `Gangs of new york`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Gangs_of_New_York_Poster.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/gangs_of_new_york.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 8.8,
  released: 2002,
  runTime: 167,
  scoresCount: 370881,
  starring: [`Leonardo DiCaprio`, `Cameron Diaz`, `Daniel Day-Lewis`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}, {
  backgroundColor: `#9B7E61`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/War_of_the_Worlds.jpg`,
  description: `As Earth is invaded by alien tripod fighting machines, one family fights for survival.`,
  director: `Steven Spielberg`,
  genre: `Adventure`,
  id: 2,
  isFavorite: false,
  name: `War of the Worlds`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/War_of_the_Worlds.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/war-of-the-worlds.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 9.3,
  released: 2005,
  runTime: 116,
  scoresCount: 386834,
  starring: [`Tom Cruise`, `Dakota Fanning`, `Tim Robbins`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}, {
  backgroundColor: `#F1E9CE`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Macbeth.jpg`,
  description: `Macbeth, the Thane of Glamis, receives a prophecy from a trio of witches that one day he will become King of Scotland. Consumed by ambition and spurred to action by his wife, Macbeth murders his king and takes the throne for himself.`,
  director: `Justin Kurzel`,
  genre: `Drama`,
  id: 3,
  isFavorite: false,
  name: `Macbeth`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Macbeth.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/macbeth.jpg`,
  previewVideoLink: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  rating: 3.3,
  released: 2015,
  runTime: 113,
  scoresCount: 48798,
  starring: [`Michael Fassbender`, `Marion Cotillard`, `Jack Madigan`],
  videoLink: `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
}, {
  backgroundColor: `#92918B`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Revenant.jpg`,
  description: `A frontiersman on a fur trading expedition in the 1820s fights for survival after being mauled by a bear and left for dead by members of his own hunting team.`,
  director: `Alejandro G. Iñárritu`,
  genre: `Action`,
  id: 4,
  isFavorite: false,
  name: `The Revenant`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Revenant.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/revenant.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 4,
  released: 2015,
  runTime: 156,
  scoresCount: 618498,
  starring: [`Leonardo DiCaprio`, `Tom Hardy`, `Will Poulter`],
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
}, {
  backgroundColor: `#F0DBA2`,
  backgroundImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/background/Johnny_English.jpg`,
  description: `After a sudden attack on the MI5, Johnny English, Britain's most confident yet unintelligent spy, becomes Britain's only spy.`,
  director: `Peter Howitt`,
  genre: `Comedy`,
  id: 5,
  isFavorite: false,
  name: `Johnny English`,
  posterImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/poster/Johnny_English.jpg`,
  previewImage: `https://htmlacademy-react-2.appspot.com/wtw/static/film/preview/johnny-english.jpg`,
  previewVideoLink: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  rating: 10,
  released: 2003,
  runTime: 88,
  scoresCount: 136843,
  starring: [`Rowan Atkinson`, `John Malkovich`, `Natalie Imbruglia`],
  videoLink: `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
}];
