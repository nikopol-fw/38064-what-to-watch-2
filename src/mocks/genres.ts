export const genres = [`All genres`, `Action`, `Adventure`, `Comedy`, `Crime`, `Drama`, `Fantasy`, `Thriller`];
