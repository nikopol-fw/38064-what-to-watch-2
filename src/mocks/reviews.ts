import {Review} from "../models/review";


export const reviews: Review[] = [{
  id: 1,
  rating: 1.3,
  comment: `Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.`,
  date: new Date(`2019-12-05T19:57:19.384Z`),
  user: {
    id: 16,
    name: `Mollie`,
  }
}, {
  id: 2,
  rating: 2.7,
  comment: `This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.`,
  date: new Date(`2019-11-30T19:57:19.384Z`),
  user: {
    id: 12,
    name: `Isaac`,
  }
}];
