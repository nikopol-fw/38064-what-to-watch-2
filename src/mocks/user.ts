import {User} from "../models/user";


export const user: User = {
  avatar: `/wtw/static/avatar/3.jpg`,
  email: `alter.ego@gmail.com`,
  id: 1,
  name: `alter.ego`,
};
