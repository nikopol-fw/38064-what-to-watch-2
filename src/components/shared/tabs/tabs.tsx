import * as React from 'react';

import {Film} from '../../../models/film';

import {TabOverview} from '../tab-overview/tab-overview';
import {TabDetails} from '../tab-details/tab-details';
import TabReviews from '../tab-reviews/tab-reviews';


enum Tab {
  OVERVIEW = `Overview`,
  DETAILS = `Details`,
  REVIEWS = `Reviews`,
}

interface Props {
  film: Film;

  activeItem: number;
  setActiveItem: (ind: number) => void;
  resetActiveItem: () => void;
}

export class Tabs extends React.PureComponent<Props> {

  constructor(props: Props) {
    super(props);

    this.props.setActiveItem(0);
  }

  private getTabClickHandler(ind: number) {
    return (evt) => {
      evt.preventDefault();
      this.props.setActiveItem(ind);
    };
  }

  render() {
    const {activeItem: activeTab, film} = this.props;
    const TabsValues = Object.values(Tab);

    return (
      <div className="movie-card__desc">
        <nav className="movie-nav movie-card__nav">
          <ul className="movie-nav__list">
            {TabsValues.map((tab, i) => (
              <li className={`movie-nav__item${activeTab === i ? ` movie-nav__item--active` : ``}`}
                key={`tab-${i}`}>
                <a href="#" className="movie-nav__link"
                  onClick={this.getTabClickHandler(i)}>{tab}</a>
              </li>
            ))}
          </ul>
        </nav>

        {(() => {
          switch (TabsValues[activeTab]) {
            case Tab.OVERVIEW:
              return (
                <TabOverview
                  description={film.description}
                  director={film.director}
                  rating={film.rating}
                  scoresCount={film.scoresCount}
                  starring={film.starring}
                />
              );

            case Tab.DETAILS:
              return (
                <TabDetails
                  director={film.director}
                  genre={film.genre}
                  released={film.released}
                  runTime={film.runTime}
                  starring={film.starring}
                />
              );

            case Tab.REVIEWS:
              return (
                <TabReviews
                  filmId={film.id}
                />
              );

            default:
              return null;
          }
        })()}
      </div>
    );
  }
}
