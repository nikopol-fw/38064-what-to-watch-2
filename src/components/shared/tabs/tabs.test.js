import React from 'react';
import renderer from 'react-test-renderer';

import {films} from '../../../mocks/films';
import {Tabs} from './tabs';


const mock = {
  film: films[0],
  activeItem: 1,
  setActiveItem: () => void (0),
  resetActiveItem: () => void (0),
};

it(`Tabs correctly renders afters relaunch`, () => {
  const tree = renderer.create(
      <Tabs
        film={mock.film}
        activeItem={mock.activeItem}
        setActiveItem={mock.setActiveItem}
        resetActiveItem={mock.resetActiveItem}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
