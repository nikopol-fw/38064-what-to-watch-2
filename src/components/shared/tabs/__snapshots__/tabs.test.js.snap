// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs correctly renders afters relaunch 1`] = `
<div
  className="movie-card__desc"
>
  <nav
    className="movie-nav movie-card__nav"
  >
    <ul
      className="movie-nav__list"
    >
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Overview
        </a>
      </li>
      <li
        className="movie-nav__item movie-nav__item--active"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Details
        </a>
      </li>
      <li
        className="movie-nav__item"
      >
        <a
          className="movie-nav__link"
          href="#"
          onClick={[Function]}
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="movie-card__text movie-card__row"
  >
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Director
        </strong>
        <span
          className="movie-card__details-value"
        >
          Martin Scorsese
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Starring
        </strong>
        <span
          className="movie-card__details-value"
          dangerouslySetInnerHTML={
            Object {
              "__html": "Leonardo DiCaprio,<br>Cameron Diaz,<br>Daniel Day-Lewis",
            }
          }
        />
      </p>
    </div>
    <div
      className="movie-card__text-col"
    >
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Run Time
        </strong>
        <span
          className="movie-card__details-value"
        >
          2h 47m
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Genre
        </strong>
        <span
          className="movie-card__details-value"
        >
          Crime
        </span>
      </p>
      <p
        className="movie-card__details-item"
      >
        <strong
          className="movie-card__details-name"
        >
          Released
        </strong>
        <span
          className="movie-card__details-value"
        >
          2002
        </span>
      </p>
    </div>
  </div>
</div>
`;
